src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/FastTrackService.cs

```diff
@@ -47,6 +47,16 @@ public async Task CreateAllByRelease(Guid releaseId, PublishContext context)
             }
         }
 
+        public async Task DeleteAllReleaseFastTracks()
+        {
+            var allPartitionKeys = (await _tableStorageService.ExecuteQueryAsync(PublicReleaseFastTrackTableName,
```

Comment:
Isn't it bit wasteful to grab all of the partition keys upfront, as I assume this will mean we scan the entire table multiple times during this delete all operation?

Reply:
It's an attempt to speed up the delete knowing that at some point we might have a lot of rows in this table (num releases x num datablocks).

The simplest option was to delete the table knowing it will be recreated again but I don't think there's a guarantee we can use the same table name straight away because the table is only soft deleted and not immediately.

This method uses 1 full table scan to retrieve a minimal version each row (partition key and row id).
It then deletes them with up to 16 parallel threads each batching the retrieval and deletion of 100 rows at a time.

Using TableBatchOperation all batches needed to have the same partition Id which means to delete all rows we still need to have retrieved them all up front.

Reply:
I see. Weird that table storage doesn't make it easier to delete things without having to employ these kinds of strategies really.

If it's not going to be an issue to be doing a full table scan to get all the partition keys. Guess i'd only be worried about thousands  of them being loaded into memory, but guess this is probably just overthinking it anyway.